File: 0019. Remove Nth Node From End of List/Solution.cpp
// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* l = head;
        ListNode* r = head;
        for (int i = 0; i < n; i++)
            l = l->next;
        if (!l) return head->next;
        while (l->next)
        {
            l = l->next;
            r = r->next;
        }
        r->next = r->next->next;
        return head;
    }
};

File: 0019. Remove Nth Node From End of List/Solution.py
# Definition for singly-linked list.
class ListNode:
   def __init__(self, val=0, next=None):
       self.val = val
       self.next = next

class Solution:
    def removeNthFromEnd(self, head, n):
        """
        type head: ListNode
        type n: int
        rtype: ListNode
        """
        l, r = head, head
        for x in range(n):
            l = l.next
        if not l: return head.next
        while l.next:
            l = l.next
            r = r.next
        r.next = r.next.next
        return head